### PR TITLE
Update osgi annotations in rest endpoint

### PR DESCRIPTION
**Pull request by plutasnyy**
I updated osgi annotations in rest endpoint

## Motivation and Context

Before, annotations was marked as deprecated

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the [code style](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

**Comment by plutasnyy** [comment]
My bad, I forgot add files to git repository ;)

File: rest-endpoint/src/main/java/com/cognifide/aet/rest/MetadataServlet.java [hunk 1]
```diff
@@ -35,19 +35,18 @@
...
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-@Service
-@Component(label = "MetadataServlet", description = "Returns Suite Metadata", immediate = true)
+@Component(service = BasicDataServlet.class, immediate = true)
```
Comment (Skejven) [inline rest-endpoint/src/main/java/com/cognifide/aet/rest/MetadataServlet.java]: Could you check if `service = HttpServlet.class` would be enough here and in other `BasicDataServlet` extensions please? We only want this class to be registered as a servlet. It won't be referenced anywhere.

File: test-executor/pom.xml [hunk 2]
```diff
@@ -123,6 +123,7 @@
 
   <build>
     <plugins>
+      <!-- ToDo: remove me when all felix dependencies are removed -->
```
Comment (Skejven) [inline test-executor/pom.xml]: Please clear also dependencies:
```
   <dependency>
      <groupId>org.apache.sling</groupId>
      <artifactId>org.apache.sling.commons.osgi</artifactId>
    </dependency>
```
and
```
    <dependency>
      <groupId>org.apache.felix</groupId>
      <artifactId>org.apache.felix.scr.annotations</artifactId>
    </dependency>
```

Comment (plutasnyy) [reply test-executor/pom.xml]: I edited wrong file, now should be correct

**Pull request merged**